version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    image: pivot-app:${TAG:-latest}
    container_name: pivot-app
    restart: unless-stopped
    volumes:
      - ./storage/app/public:/var/www/storage/app/public
    environment:
      - APP_ENV=production
      - APP_DEBUG=false
      - DB_CONNECTION=mysql
      - DB_HOST=mysql
      - DB_PORT=3306
      - DB_DATABASE=pivot
      - DB_USERNAME=root
      - DB_PASSWORD=8ocxlumnakezr2wdfcwiijct2rejsgdr
    networks:
      - pivot-network
      - traefik-network
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.pivot.rule=Host(`pivot.guillaume-lcte.fr`)"
      - "traefik.http.routers.pivot.entrypoints=websecure"
      - "traefik.http.routers.pivot.tls.certresolver=letsencrypt"
      - "traefik.http.services.pivot.loadbalancer.server.port=4004"
      - "traefik.docker.network=traefik-network"

  mysql:
    image: mysql:8.0
    container_name: pivot-mysql
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: pivot
      MYSQL_ROOT_PASSWORD: 8ocxlumnakezr2wdfcwiijct2rejsgdr
    volumes:
      - mysql-data:/var/lib/mysql
    networks:
      - pivot-network

networks:
  pivot-network:
    driver: bridge
  traefik-network:
    external: true

volumes:
  mysql-data:
    driver: local 